lib/mail.ex

```diff
@@ -312,6 +312,30 @@ defmodule Mail do
     renderer.render(message)
   end
 
+  @doc """
+  Get messages containing attachments
+
+  Argument must be a `Mail.Message` or `List` of `Mail.Message`
+
+  Return a `List` of `Mail.Message` or emty `List`
+  """
+  def get_attachments(parts, attachments \\ [])
```

Comment:
Ok. I did this for my own case, I think you know better than me all use cases and how to implement it.